File: src/pcsalt/example/clipboardmanager/MainActivity.java
package pcsalt.example.clipboardmanager;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.ClipData;
import android.content.Context;
import android.os.Bundle;
import android.view.Menu;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends Activity {

	EditText etMessage;
	Button btnCopy;
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		etMessage	= (EditText) findViewById(R.id.etMessage);
		btnCopy 	= (Button)	 findViewById(R.id.btnCopy);
		
		// on button click copy text in EditText to clipboard
		btnCopy.setOnClickListener(new OnClickListener() {
			
			@SuppressLint("NewApi")
			@Override
			public void onClick(View arg0) {
				String text = etMessage.getText().toString().trim();
				if(text.length() > 0) {
					if(android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.HONEYCOMB) {
						@SuppressWarnings("deprecation")
						android.text.ClipboardManager clipboardMgr = (android.text.ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);
						clipboardMgr.setText(text);
					} else {
						// this api requires SDK version 11 and above, so suppress warning for now
						android.content.ClipboardManager clipboardMgr = (android.content.ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);
						ClipData clip = ClipData.newPlainText("Copied text", text);
						clipboardMgr.setPrimaryClip(clip);
					}
				}
			}
		});
		
		
	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		// Inflate the menu; this adds items to the action bar if it is present.
		getMenuInflater().inflate(R.menu.main, menu);
		return true;
	}

}

File: app/src/main/kotlin/com/pcsalt/example/clipboardmanager/MainActivity.kt
package com.pcsalt.example.clipboardmanager

import android.content.ClipData
import android.content.ClipboardManager
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.pcsalt.example.clipboardmanager.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

  private lateinit var binding: ActivityMainBinding

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    binding = ActivityMainBinding.inflate(layoutInflater)
    setContentView(binding.root)

    val clipboard = getSystemService(CLIPBOARD_SERVICE) as ClipboardManager

    binding.btnCopy.setOnClickListener {
      val text = binding.etInput.text.toString().trim()
      if (text.isEmpty()) {
        binding.etInput.error = "Enter text to copy"
        return@setOnClickListener
      }
      val clip = ClipData.newPlainText("copied_text", text)
      clipboard.setPrimaryClip(clip)
      Toast.makeText(this, "Copied to clipboard", Toast.LENGTH_SHORT).show()
    }

    binding.btnPaste.setOnClickListener {
      val clip = clipboard.primaryClip
      if (clip != null && clip.itemCount > 0) {
        val pastedText = clip.getItemAt(0).text.toString()
        binding.tvPastedText.text = pastedText
      } else {
        Toast.makeText(this, "Clipboard is empty", Toast.LENGTH_SHORT).show()
      }
    }
  }
}
